viewlog.c

```diff
@@ -54,13 +57,20 @@ void ViewLog(int config)
 
   /* Try first using file association */
   CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE); /* Safe to init COM multiple times */
-  status = ShellExecuteW (o.hWnd, L"open", o.conn[config].log_path, NULL, o.log_dir, SW_SHOWNORMAL);
-
+  if (AssocQueryString(0, ASSOCSTR_EXECUTABLE, L".log", NULL, assocexe, &assocexe_num) == S_OK)
```

Comment:
This will mis-behave (most of the time fails, could segfault) unless assocexe_num is set to the size of the buffer:
assocexe_num = _countof(assocexe)